service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -128,17 +116,23 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
         registerAdditionalResources();
 
         final Future<Void> result = Future.future();
-        CompositeFuture.all(bindSecureHttpServer(), bindInsecureHttpServer())
-                .compose(r -> result.complete(), result);
+        bindSecureHttpServer().setHandler(res -> {
+            if (res.failed()){
+                bindInsecureHttpServer().compose(r -> result.complete(), result);
+            } else if (config.isInsecurePortEnabled() && config.isSecurePortEnabled()) {
+                bindInsecureHttpServer().compose(r -> result.complete(), result);
```

Comment:
this binds the insecure server to the same port as the secure server, right? What is the purpose of this? Does this even work? IMHO the insecure server should only be started in this case (i.e. the secure server has been started already) if the insecure port is different from the secure one ...

Reply:
that what I  wanted to do in the first place, but I tried this and it seems to work, but I there was a environment variable in my setup that caused me to misunderstand the result. I'll push a fix right now

Reply:
@sophokles73  now it should be good.

Reply:
I think this is still not correct. The user can indicate that the insecure server should be started by doing any or both of the following:
* set *insecurePortEnabled* to `true`
* set *insecurePort* to a value other than `Constants.PORT_UNCONFIGURED`

If only *insecurePortEnabled* is set to `true` but no specific port is set, then the default insecure port is used.
IMHO this means for the HealthCheckServer, assuming that the secure server has started successfully:
only start the insecure server
* if *insecurePortEnabled* is set to `true` and *config.getInsecurePort(DEFAULT_PORT)* is different from *server.actualPort()*, or
* if *insecurePort* is not `Constants.PORT_UNDEFINED` and *config.getInsecurePort()* is different from *server.actualPort()*

Reply:
Any suggestion for the insecure port default value ? What about `8043` ?

Reply:
I think we can still use the same default port. If somebody wants to use a secure **and** insecure server, they should be required to select either a specific secure and/or insecure port ...

Reply:
:+1: